macros/src/main/scala/com/softwaremill/macwire/dependencyLookup/ValuesOfTypeInEnclosingClassFinder.scala

```diff
@@ -102,6 +229,16 @@ private[dependencyLookup] class ValuesOfTypeInEnclosingClassFinder[C <: Context]
     }
 
     debug("Looking in the enclosing class/trait")
-    doFind(enclosingClassBody, Nil)
+
+    // Create a special Set that uses structural equality on Tree;
+    // `equals` on trees is reference equality.
+    val emptyTreeSet = TreeSet.empty(new Ordering[Tree] {
+      override def compare(x: c.universe.Tree, y: c.universe.Tree): Int = {
+        if (x.equalsStructure(y)) 0
+        else x.hashCode().compareTo(y.hashCode()) // some arbitrary comparison that should retain the compare semantic
```

Comment:
Well then that's fine, because that's `hashcode()` contract right? If they are equal by reference they are logically equal structurally (reference equality being stronger that structural equality). So I really doubt that given 2 trees t1 and t2 you can have both:
```
t1.hashCode() == t2.hashCode()
t1.equalsStructure(t2) == false
```